### PR TITLE
Use Yarn package manager

### PR DESCRIPTION
**Pull request by stylemistake**
## About The Pull Request

Ok, so I am a little bit disappointed in `npm`, because we constantly have package instability, and builds are failing because someone has different packages.

- `package-lock.json` is constantly updating all over the place even when dependencies don't change, making it practically useless for locking dependency versions.
- To support a multi-package setup we had to use `lerna` which makes `package-lock.json` even less deterministic, and required us to do a few dirty hacks with symlinks to make it all work.

What do we get with Yarn:

- `yarn.lock` is actually good, and ensures reproducible package installs.
- A proper, out of the box support for a multi-package setup.

I have used Yarn for more than 2 years at work, deploying many critical systems at scale, and it has never failed me with unpredictable installs.

Note that we have initially used Yarn for tgui-next, but MSO complained about it, because `yarn.lock` file looks like a temporary file and we didn't want to burden devs with a yet another hard dependency.

Npm promised that one day it will [support workspaces and deterministic lock files](https://blog.npmjs.org/post/186983646370/npm-cli-roadmap-summer-2019), but nobody knows when that will happen, it's probably years behind the schedule.

## Why It's Good For The Game

Improved developer experience, at a small one-time cost of installing [Yarn](https://yarnpkg.com/en/docs/install) on your system.

## Changelog
:cl:
fix: Replaced npm with yarn for more deterministic installs in tgui-next.
/:cl:

### CODE REVIEW

**Comment by stylemistake** [comment]
Travis build fixed, needs review.

**Comment by stylemistake** [comment]
`@spookydonut` (NodeJS 12.13, Windows) and `@willox` (NodeJS 12.10, WSL) have tested dirty installs on top of their own feature branches.

> I hope you have a new package manager lined up for next week?

I do not plan on replacing Yarn.

**Comment by actioninja** [comment]
I'm really not a big fan of requiring another dependency. Normally dependencies aren't a huge deal in software dev, but ideally the only thing you should need to dev is byond itself. Really disappointed that it may have to come to requiring two external dependencies. I've already expressed a ton of these concerns relating to long term dev overhead, so I'm not going to repeat them here.

However, I will say consistently and always functioning, and not breaking horribly in unexpected ways will be a lot better for maintainability and development in the long run. I hate that it had to come to this, but a functional and stable environment is a lot more important than an easy to setup one. They are somewhat related, but we really need to make sure that this is going to work when we need it to. The ship of "no outside dependencies" sailed long ago on original tgui long ago so requiring one extra dependency isn't completely unprecedented, but at the same time "oh come on, it's just _one_ more package" is a very real slippery slope.

In other words, I really don't like this but I also recognize it may be the only sane path forward that isn't going to lead to major headaches down the line.

If you could figure out some way to get CI to automate the build process, this wouldn't be as much of an issue.

**Review by actioninja — APPROVED** [review]
Tested and it works fine.

**Comment by spookydonut** [comment]
Yeah this fixed building/watching for me.

**Comment by spookydonut** [comment]
You already have to install npm/node if you want to work on your own tgui so who gives a shit if you just install yarn too.

**Comment by actioninja** [comment]
This is currently blocking and I think a lot of the meaningful discussion to come out of this already has.
If this ends up being a huge issue later down the line, that's what git is for.
